### ISSUE TITLE
Double quotes in `$@` combined with `printf` is breaking the code.

### ISSUE THREAD

**Issue by droslean** [koalaman/shellcheck#1408]
#### For bugs
- Rule Id SC2068
- My shellcheck version:
```console
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.5.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net

```
- [ X ] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [ X ] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
version() { IFS="."; printf "%03d%03d%03d\\n" $@; }
version "1.8.1"
```
The only way for this code to work is not to have the `$@` in double quotes.
Otherwise:
```
 printf: 1.8.1: invalid number
```

But shellcheck is still complains to keep the `$@` in double quotes, and that will break my code.

And in order to make my code work and keep the `shellcheck` satisfy, I will have to do some really ugly stuff to parse the variable and split it etc......

`Shellcheck` should allow that in `printf` to pass the array without the double quotes.

**Comment by koalaman** [comment]
ShellCheck is not always right. Therefore, you can selectively [ignore issues](https://github.com/koalaman/shellcheck/wiki/Ignore) using special comment directives:

    version() {
       IFS=".";
       # shellcheck disable=SC2068
       printf "%03d%03d%03d\\n" $@;
    }

However, it might arguably be better rewritten as:

    version() { printf '%s' "$1" | { IFS="." read -r a b c _; printf '%03d' "$a" "$b" "$c"; } }

which behaves better for `'1.8.0.1'`, `'1./*/*/*.2'`, and which doesn't leave IFS in a confused state if you ever call it outside of a subshell. 

This is absolutely not a general case where `$@` should be allowed unquoted without question, so ShellCheck is right to warn.
